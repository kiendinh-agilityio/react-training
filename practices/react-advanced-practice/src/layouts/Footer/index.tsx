// Import constants
import { FOOTER_NAVBAR } from '@/constants';

// Import utils
import { currentYear } from '@/utils';

// Import components
import { Link } from '@/components/common';

const Footer = () => (
  <footer className="flex justify-between text-xs text-base px-[21px]">
    <p>
      @ {currentYear}, Made with ❤️ by{' '}
      <span className="text-primary font-bold">Creative Tim</span> &{' '}
      <span className="text-primary font-bold">Simmmple</span> for a better web
    </p>
    <nav>
      <ul className="flex gap-11">
        {FOOTER_NAVBAR.map((item) => (
          <li key={item.href}>
            <Link href={item.href} target="_blank">
              {item.label}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  </footer>
);

export default Footer;
