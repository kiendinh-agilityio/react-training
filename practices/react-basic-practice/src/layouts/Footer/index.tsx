// Import constants
import { NAVBAR_ITEMS } from '@/constants';

// Import utils
import { currentYear } from '@/utils';

// Import components
import { Link } from '@/components/common';

const Footer = () => (
  <footer className="flex justify-between text-sidebar px-[21px]">
    <p>
      @ {currentYear}, Made with ❤️ by{' '}
      <span className="text-primary font-helveticaBold font-bold">Creative Tim</span> &{' '}
      <span className="text-primary font-helveticaBold font-bold">Simmmple</span> for a better web
    </p>
    <nav>
      <ul className="flex gap-11">
        {NAVBAR_ITEMS.map((item) => (
          <li key={item.href}>
            <Link href={item.href} target="_blank" text={item.label} />
          </li>
        ))}
      </ul>
    </nav>
  </footer>
);

export default Footer;
