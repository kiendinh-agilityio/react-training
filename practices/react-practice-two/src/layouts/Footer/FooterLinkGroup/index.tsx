// Import common components
import { Link, Paragraph } from '@/components/common';

// Import types
import { ParagraphVariant, TextSize } from '@/types';

// import components
import { SocialList } from '@/components';

interface LinkItem {
  id: string;
  href: string;
  label: string;
}

interface FooterLinkGroupProps {
  title: string;
  items: LinkItem[];
  hasSocialLinks?: boolean;
}

const FooterLinkGroup = ({ title, items, hasSocialLinks = false }: FooterLinkGroupProps) => {
  const renderFooterLinkGroup = items.map((item) => (
    <li key={item.id}>
      <Link href={item.href} target="_blank" text={item.label} />
    </li>
  ));

  return (
    <div className="text-light">
      <Paragraph
        variant={ParagraphVariant.Bold}
        size={TextSize.Medium}
        text={title}
        className="leading-[25px]"
      />
      <ul className="mt-[35px]">{renderFooterLinkGroup}</ul>
      {hasSocialLinks && <SocialList />}
    </div>
  );
};

export default FooterLinkGroup;
