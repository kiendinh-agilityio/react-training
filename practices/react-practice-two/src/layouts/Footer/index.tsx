// import FooterLinkGroup
import FooterLinkGroup from './FooterLinkGroup';

// import components
import { SocialList, Paragraph } from '@/components/common';

// import image
import { FooterImage } from '@/assets/images';

// import constants
import { ABOUT_LIST, RESOURCES_LIST, LEGALS_LIST } from '@/constants';

// import utils
import { currentYear } from '@/utils';

// import types
import { TextSize } from '@/types';

const Footer = () => (
  <footer className="container bg-primary pt-[99px] px-[196px] pb-[30px] relative flex flex-col justify-end">
    <img src={FooterImage} alt="Footer Banner" className="absolute left-0 bottom-16" />
    <div className="flex flex-col items-end">
      <div className="flex justify-between gap-[113px]">
        <FooterLinkGroup title="About" items={ABOUT_LIST} />
        <FooterLinkGroup title="Resources" items={RESOURCES_LIST} />
        <FooterLinkGroup title="Legals" items={LEGALS_LIST} />
      </div>
      <SocialList />
    </div>
    <Paragraph
      size={TextSize.Small}
      text={`Â© Copyright ${currentYear}. All rights reserved.`}
      className="text-light leading-[24px] text-center mt-[71px]"
    />
  </footer>
);

export default Footer;
