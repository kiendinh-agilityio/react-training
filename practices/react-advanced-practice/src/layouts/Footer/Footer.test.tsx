import { render } from '@testing-library/react';
import Footer from '.';

describe('Footer Layouts', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
