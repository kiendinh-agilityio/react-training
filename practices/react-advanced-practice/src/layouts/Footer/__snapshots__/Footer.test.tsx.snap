// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Layouts should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="flex justify-between text-base px-[21px]"
  >
    <p>
      @ 2025, Made with ❤️ by 
      <span
        class="text-primary font-bold"
      >
        Creative Tim
      </span>
       & 
      <span
        class="text-primary font-bold"
      >
        Simmmple
      </span>
       for a better web
    </p>
    <nav>
      <ul
        class="flex gap-11"
      >
        <li>
          <a
            class="rt-Text rt-reset rt-Link rt-underline-auto cursor-pointer undefined"
            data-accent-color=""
            href="https://www.creative-tim.com"
            target="_blank"
          >
            Creative Tim
          </a>
        </li>
        <li>
          <a
            class="rt-Text rt-reset rt-Link rt-underline-auto cursor-pointer undefined"
            data-accent-color=""
            href="https://www.simmmple.com"
            target="_blank"
          >
            Simmmple
          </a>
        </li>
        <li>
          <a
            class="rt-Text rt-reset rt-Link rt-underline-auto cursor-pointer undefined"
            data-accent-color=""
            href="/blog"
            target="_blank"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            class="rt-Text rt-reset rt-Link rt-underline-auto cursor-pointer undefined"
            data-accent-color=""
            href="/license"
            target="_blank"
          >
            License
          </a>
        </li>
      </ul>
    </nav>
  </footer>
</DocumentFragment>
`;
